test/sitetype.jl

```diff
@@ -15,13 +15,62 @@ using ITensors, Test
     @test SySy ≈ product(Sy, Sy)
 
     sites = siteinds("S=1", N)
-    @test_throws ArgumentError op(sites, "Sp", 1)
+    #@test_throws ArgumentError op(sites, "Sp", 1)
     Sz = op(sites, "Sz", 2)
     SzSz = op(sites, "Sz * Sz", 2)
     @test SzSz ≈ product(Sz, Sz)
     Sy = op(sites, "Sy", 2)
     SySy = op(sites, "Sy * Sy", 2)
     @test SySy ≈ product(Sy, Sy)
+    SzSySz = op(sites, "Sz * Sy * Sz", 2)
+    @test SzSySz ≈ product(Sz, product(Sy, Sz))
+  end
+
+  @testset "+/- in operator strings" begin
+    q = siteind("Qudit"; dim=5)
+    Amat = array(op("a", q))
+    Adagmat = array(op("a†", q))
+
+    x = Amat - Adagmat
+    @test x ≈ array(op("a - a†", q))
+    x = Amat * Adagmat - Adagmat
+    @test x ≈ array(op("a * a† - a†", q))
+    @test x ≈ array(op("a*a† - a†", q))
+    x = Adagmat * Adagmat * Amat * Amat
+    @test x ≈ array(op("a† * a† * a * a", q))
+
+    q = siteind("S=1/2")
+    Sp = array(op("S+", q))
+    Sm = array(op("S-", q))
+    Sx = array(op("Sx", q))
+    Sy = array(op("Sy", q))
+    Sz = array(op("Sz", q))
+    x = Sp + Sm
+    @test x ≈ array(op("S+ + S-", q))
+    x = Sp - Sm
+    @test x ≈ array(op("S+ - S-", q))
+    x = Sp - Sm - Sp
+    @test x ≈ array(op("S+ - S- - S+", q))
+    x = Sp * Sm + Sm * Sp
+    @test x ≈ array(op("S+ * S- + S- * S+", q))
+    @test x ≈ array(op("S+*S- + S-*S+", q))
```

Comment:
We should be able to test that it outputs a deprecation warning, for example: https://discourse.julialang.org/t/help-with-deprecation-warnings-in-tests/50011